dojo/tools/npm_audit_7_plus/parser.py

```diff
@@ -0,0 +1,237 @@
+            items[unique_key] = item
+
+        return list(items.values())
+
+
+def get_item(item_node, tree, test):
+    """Return the individual Findigns from items found in report."""
+    vuln_ids = []
+    references = []
+    mitigation = ""
+    test = test
+    static_finding = True
+    title = ""
+    unique_id_from_tool = ""
+    cvssv3 = ""
+    cwe = ""
+
+    if item_node["severity"] == "low":
+        severity = "Low"
+    elif item_node["severity"] == "moderate":
+        severity = "Medium"
+    elif item_node["severity"] == "high":
+        severity = "High"
+    elif item_node["severity"] == "critical":
+        severity = "Critical"
+    else:
+        severity = "Info"
+
+    if item_node["via"] and isinstance(item_node["via"][0], str):
+        # this is a top level key (a vulnerability)
+        title = item_node["name"]
+        cwe = "CWE-1035"  # default
+        component_name = title
+        component_version = "n/a"
+
+    elif item_node["via"] and isinstance(item_node["via"][0], dict):
+        title = item_node["via"][0]["title"]
+        component_name = item_node["nodes"][0]
+        component_version = item_node["range"]
+        cwe = item_node["via"][0]["cwe"][0]
+        vuln_ids.append(cwe)
+        references.append(item_node["via"][0]["url"])
+        unique_id_from_tool = str(item_node["via"][0]["source"])
+        cvssv3 = item_node["via"][0]["cvss"]["vectorString"]
+
+    if isinstance(item_node["fixAvailable"], dict):
+        fix_name = item_node["fixAvailable"]["name"]
+        fix_version = item_node["fixAvailable"]["version"]
+        mitigation = "Update {0} to version {1}".format(fix_name, fix_version)
+    else:
+        mitigation = "No specific mitigation provided by tool."
+
+    description = get_vuln_description(item_node, tree)
+
+    if (item_node["via"] and
+        isinstance(item_node["via"][0], dict) and
+            len(item_node["via"]) > 1):
+        # we have a multiple CWE vuln which we will capture in the
+        # vulnerability_ids and references
+        for vuln in item_node["via"][1:]:  # have to decide if str or object
+            if isinstance(vuln, dict):
+                vuln_ids.append(vuln["cwe"][0])
+                references.append(vuln["url"])
+
+    if len(cwe):
+        cwe = int(cwe.split("-")[1])
+
+    dojo_finding = Finding(
+        title=title,
+        test=test,
+        severity=severity,
+        description=description,
+        cwe=cwe,
+        mitigation=mitigation,
+        references=", ".join(references),
+        component_name=component_name,
+        component_version=component_version,
+        false_p=False,
+        duplicate=False,
+        out_of_scope=False,
+        mitigated=None,
+        impact="No impact provided",
+        static_finding=static_finding,
+        dynamic_finding=False,
+        unique_id_from_tool=unique_id_from_tool,
+        vuln_id_from_tool=unique_id_from_tool
```

Comment:
I think we usually only populate one of these with the same identifier. I tried to dig into what `source` actually means to `npm audit`, and as far as I can tell it is a stable identifier for either a CVE or GHSA, not an individual *instance* of an issue detected in a specific codebase. So I think it better [matches the intent of `vuln_id_from_tool`](https://github.com/DefectDojo/django-DefectDojo/blob/82181253cd303efd1642994327e1d862c6b247ef/dojo/models.py#L2444-L2453)

```suggestion
        vuln_id_from_tool=unique_id_from_tool
```